subsys/net/lib/shell/iface.c

```diff
@@ -817,6 +818,73 @@ static int cmd_net_default_iface(const struct shell *sh, size_t argc, char *argv
 	return 0;
 }
 
+static int cmd_net_link_speed(const struct shell *sh, size_t argc, char *argv[])
+{
+	int idx = get_iface_idx(sh, argv[1]);
+	const struct device *phy_dev;
+	bool half_duplex = false;
+	struct net_if *iface;
+	uint16_t speed = 0;
+
+	if (argc < 3) {
+		PR_WARNING("Usage: net iface set_link <index>"
+			   "<Speed:10/100/1000/2500/5000> <Duplex[optional]:h/f>\n");
+		return -ENOEXEC;
+	}
+
+	iface = net_if_get_by_index(idx);
+	if (net_if_l2(iface) != &NET_L2_GET_NAME(ETHERNET)) {
+		PR_WARNING("Interface %d is not Ethernet type\n", idx);
+		return -EINVAL;
+	}
+
+	phy_dev = net_eth_get_phy(iface);
+	if (!phy_dev) {
+		PR_WARNING("No PHY device found for interface %d\n", idx);
+		return -ENODEV;
+	}
+
+	for (int k = 2; k < argc; k++) {
+		/* Reset for each speed argument */
+		half_duplex = false;
+		if ((k + 1 < argc) && (argv[k+1][0] == 'h')) {
+			half_duplex = true;
+		}
```

Comment:
```suggestion
		if ((k + 1 < argc) && (argv[k+1][0] == 'h')) {
			half_duplex = true;
		} else {
			half_duplex = false;
		}
```

put false in the else